### ISSUE TITLE
Angular CLI 12 generates a project with 35 npm vulnerabilities

### ISSUE THREAD

**Issue by hardmaster92** [angular/angular-cli#20795]
After installing the latest CLI (v12) I decided to generate a clean new project to see what's different and what's being generated. Unfortunately I found out that the generated project has 35 severity vulnerabilities in the npm packages right off the bat. The vulnerabilities are not critical but they can't be resolved automatically using `npm audit fix`. The audit report shows all the vulnerabilities in the following way:
```
Moderate        Regular Expression Denial of Service

Package         postcss

Patched in      >=8.2.10

Dependency of   @angular-devkit/build-angular [dev]

Path            @angular-devkit/build-angular > postcss-preset-env >
                    autoprefixer > postcss
```

I hope that I reported the issue in the correct repository.

**Comment by alan-agius4** [comment]
Blocked on https://github.com/csstools/postcss-preset-env/issues/191

**Comment by crooksey** [comment]
Probably the best place for this, with the current postcss-present-env package angular is also using PostCSS v7 and v8, which has its own errors:

```
Unknown error from PostCSS plugin. Your current PostCSS version is 8.2.14, but postcss-preset-env uses 7.0.35. Perhaps this is the source of the error below.
```

When running the following command:

```
npm ls postcss 
```

I can see that postCSS V8 is used in:

- css-loader@5.2.4
- cssnano@5.0.2
- postcss-import@14.0.1
- postcss-loader@5.2.0

Then the following dependencies use V7:

- postcss-preset-env@6.7.0

**Comment by hfournier** [comment]
Seeing the same thing in Angular 11.2.14 project, but following a different path to postcss:
```
  Moderate        Regular Expression Denial of Service                          

  Package         postcss                                                       

  Patched in      >=8.2.10                                                      

  Dependency of   @angular-devkit/build-angular [dev]                           

  Path            @angular-devkit/build-angular > resolve-url-loader > postcss  

  More info       https://npmjs.com/advisories/1693     
```
running `npm ls postcss`, I get:
```
+-- @angular-devkit/build-angular@0.1102.13
| +-- css-loader@5.0.1
| | `-- UNMET PEER DEPENDENCY postcss@8.2.15
| +-- UNMET PEER DEPENDENCY postcss@8.2.14
| `-- resolve-url-loader@4.0.0
|   `-- postcss@7.0.35
+-- netlify-cli@3.30.2
| `-- @netlify/zip-it-and-ship-it@4.0.1
|   `-- precinct@6.3.1
|     `-- detective-postcss@3.0.1
|       `-- postcss@7.0.35
`-- netlify-schematics@3.2.0
  `-- netlify-cli@2.71.0
    `-- @netlify/zip-it-and-ship-it@2.1.1
      `-- precinct@6.3.1
        `-- detective-postcss@3.0.1
          `-- postcss@7.0.32
```

**Comment by jimmyengman** [comment]
I have updated a project to angular 12.0.1 and when running `npm ls postcss` I get the same result as @crooksey with the addition that also resolve-url-loader@4.0.0 uses postcss@7.0.35

The following dependencies use V7:

- postcss-preset-env@6.7.0
- resolve-url-loader@4.0.0

**Comment by crooksey** [comment]
@jimmyengman you shouldn't need resolve-url-loader now as it is a dependency of  @angular-devkit/build-angular@12.0.1

**Comment by hfournier** [comment]
Beyond the vulnerabilities mentioned above, now I find I can no longer build and deploy my project. Angular 11.2.14, Node 14.17.0
I get:
```
An unhandled exception occurred: PostCSS plugin postcss-discard-comments requires PostCSS 8.
```
running:
```
npm ls postcss-discard-comments
```
produces:
```
`-- @angular-devkit/build-angular@0.1102.13
  `-- cssnano@5.0.2
    `-- cssnano-preset-default@5.1.1
      `-- postcss-discard-comments@5.0.1
```

**Comment by alan-agius4** [comment]
@hfournier, your issue is unrelated to this, please file a new one with a minimal reproduction. Thanks.

**Comment by hfournier** [comment]
@alan-agius4 I can do that, but they're both related to the need to upgrade to PostCSS 8.

**Comment by alan-agius4** [comment]
`postcss-discard-comments` is a dependency of cssnano which already uses postcss 8.

**Comment by hfournier** [comment]
> 
> 
> `postcss-discard-comments` is a dependency of cssnano which already uses postcss 8.

see my previous post... resolve-url-loader@4.0.0 is using postcss@7.0.35

**Comment by pascalm19** [comment]
@angular-devkit/build-angular version 0.1102.14 
A yarn audit returns that it has 3 vulnerabilities: 2 moderate | 1 high
glob-parent : Patched in >=5.1.2  
postcss : Patched in >=8.2.10
css-what : Patched in >=5.0.1

**Comment by SymbioticKilla** [comment]
glob-parent: https://github.com/angular/angular-cli/issues/21097
css-what: https://github.com/svg/svgo/issues/1488

**Comment by alan-agius4** [comment]
Closing as the original reported issue should be addressed now since postcss released a security fix in 7.0.36.

More info: https://npmjs.com/advisories/1693
